.github/workflows/integration-testing.yml

```
@@ -59,7 +67,10 @@ jobs:
       - name: Build and Push Docker image
         uses: docker/build-push-action@v2
         with:
-          push: true
+          push: ${{ github.event.inputs.push == 'true' || 
+            github.event_name == 'schedule' ||
+            ( github.event_name == 'push' && github.ref == 'refs/heads/main' )
+            }}
```

Not sure if you think it is necessary to check for the `event_type == workflow_dispatch`

```suggestion
          push: ${{
            github.event_name == 'schedule' ||
            ( github.event_name == 'push' && github.ref == 'refs/heads/main' ) ||
            ( github.event_name == 'workflow_dispatch' && github.event.inputs.push == 'true' )
            }}
```